### PR TITLE
fix(rest): set statusCode as 201 for POST request

### PR DESCRIPTION
**Pull request by virkt25**
temporary fix for #788

## Checklist

- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

### CODE REVIEW

File: packages/rest/src/rest.server.ts [hunk 1]
```diff
@@ -197,6 +197,11 @@ export class RestServer extends Context implements Server, HttpServerLike {
       return Promise.resolve();
     }
 
+    // Any item down the sequence chain can override the statusCode if needed
+    if (request.method === 'POST') {
```
Comment (raymondfeng) [inline packages/rest/src/rest.server.ts]: This is too hacky. Not all `POST` is for `create`. Can we allow/infer the statusCode from https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#responses-object?

Comment (raymondfeng) [reply packages/rest/src/rest.server.ts]: There are a few potential options for controller methods to express status code:
1. Direct access to `Response` object
2. Decorate the method to customize the default status code
3. Return a `HttpResponse` object such as `{statusCode: 201, headers: {}, body: result}`

Comment (virkt25) [reply packages/rest/src/rest.server.ts]: We have issues to track the stories of providing a way for users to set the statusCode and appropriate response headers. 

I personally like option 3 -- but for a `POST` request I think we should automatically be defaulting to 201 if a different statusCode is not provided.

**Comment by virkt25** [comment]
I believe @jannyHou is investigating a less "hacky" approach so I'll close this PR for now.

File: packages/rest/src/rest.server.ts (hunk 1 above)
Comment (jannyHou) [reply packages/rest/src/rest.server.ts]: @virkt25 After I did my PoC in https://github.com/strongloop/loopback-next/pull/1260, I tend to agree with your workaround here as the quickest fix....or allowing more time, we can do 3.

Miroslav brings an important thing: when a response returns `201`, it also includes a header with the URI of created item. 

> but for a POST request I think we should automatically be defaulting to 201 if a different statusCode is not provided.

Then the user need to provide `header` in `result` by default as well.

**Pull request closed without merging**